modoboa/admin/management/commands/subcommands/_import.py

```diff
@@ -52,20 +53,23 @@ def _import(self, filename, options):
                 progressbar.Percentage(), progressbar.Bar(), progressbar.ETA()
             ], maxval=num_lines
         ).start()
-        with open(filename) as f:
+        with io.open(filename, encoding="utf8") as f:
             reader = csv.reader(f, delimiter=options["sepchar"])
             i = 0
             for row in reader:
                 if not row:
                     continue
-                try:
-                    fct = getattr(lib, "import_%s" % row[0].strip())
-                except AttributeError:
```

Comment:
Can you explain why this way of looking for import function was wrong please?

Reply:
`row[0]` is taken directly from the csv file with no checks on what it contains, there's a potential for code injection, I'm not sure how vulnerable python is to this.

Also Python 2 didn't like it when I change domain to d**ó**main in one of the csv file import tests because Python 2 doesn't allow unicode characters in method names.

The other option is to register the standard imports with `admin_signals.import_object` in `admin/handlers.py` the same way as extensions do.
```python
@receiver(admin_signals.import_object)
def get_import_func(sender, objtype, **kwargs):
    """Return function used to import objtype."""
    if objtype == "domain":
        return lib.import_domain
    elif objtype == "domainalias":
        return lib.import_domainalias
    elif objtype == "account":
        return lib.import_account
    elif objtype == "alias":
        return lib.import_alias
    elif objtype == "forward":
        return lib.import_forward
    elif objtype == "dlist":
        return lib.import_dlist
    return None
```
And change this:
https://github.com/modoboa/modoboa/blob/2c4fefca02d1591a847de835b2974e7f6874a509/modoboa/admin/management/commands/subcommands/_import.py#L61-L68
to:
```python
fct = signals.import_object.send(
    sender=self.__class__, objtype=row[0].strip())
if not fct or fct[0][1] is None:
    continue
fct = fct[0][1]
```

Reply:
I don't think python is vulnerable to this but I'm not sure... Anyway, I do prefer this new approach :)

Reply:
fixed